storage/interface.go

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2022 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package storage
+
+import (
+	"github.com/nuts-foundation/go-stoabs"
+	"github.com/nuts-foundation/nuts-node/core"
+)
+
+// Engine defines the interface for the storage engine.
+type Engine interface {
+	core.Engine
+	core.Configurable
+	core.Runnable
+	Provider
+}
+
+// Provider lets callers get access to stores.
+type Provider interface {
+	// GetIterableKVStore returns a key-value store like KVStore, but then iterable (supporting cursors).
```

Comment:
We might have to assume not all implementations can iterate "in order".... So rewalking the DAG is actually reading values by bulk (since LC keys are calculated in advance).....

Is iterate still needed.

Reply:
Waiting for https://github.com/nuts-foundation/go-stoabs/pull/3